commons-business-impl/src/main/java/com/appjars/saturn/service/validation/DeletionValidator.java

```diff
@@ -37,4 +40,10 @@
 		};
 	}
 
+	static <T> DeletionValidator<T> forCondition(Predicate<T> predicate, Function<T, ErrorDescription> errorSupplier) {
+		Objects.requireNonNull(predicate);
```

Comment:
Given that all `forCondition` implementations are practically the same, is it possible to reuse the code in a static method in Validator interface?

Reply:
No, because the type of the lambda expression is derived from the method return type and `Validator.forCondition` returns `Validator<T>`

```
static <T> DeletionValidator<T> forCondition(Predicate<T> predicate, Function<T, ErrorDescription> errorSupplier) {
   return Validator.forCondition(predicate,errorSupplier);
}
```

What would work (at the expense of introducing an unnecessary proxy which can be avoided by this duplication) is:
```
static <T> DeletionValidator<T> forCondition(Predicate<T> predicate, Function<T, ErrorDescription> errorSupplier) {
  return (DeletionValidator<T>) Validator.<T>on(ServiceValidationKind.DELETE).forCondition(predicate, errorSupplier);
}
```

Reply:
No, if this is such a hassle, I think it's better to leave it as it is right now. We're only saving three lines of code that are perfectly readable. Disregard.